AnnService/src/Core/CommonDataStructure.cpp

```diff
@@ -3,130 +3,102 @@
 
 #include "inc/Core/CommonDataStructure.h"
 
-using namespace SPTAG;
+namespace SPTAG
+{ 
 
 const ByteArray ByteArray::c_empty;
 
-ByteArray::ByteArray()
-    : m_data(nullptr),
-    m_length(0)
+ByteArray::ByteArray() noexcept
+    : m_data(nullptr)
+    , m_length(0)
 {
 }
 
-
-ByteArray::ByteArray(ByteArray&& p_right)
-    : m_data(p_right.m_data),
-    m_length(p_right.m_length),
-    m_dataHolder(std::move(p_right.m_dataHolder))
+ByteArray::ByteArray(ByteArray&& p_right) noexcept
+    : m_data(p_right.m_data)
+    , m_length(p_right.m_length)
+    , m_dataHolder(std::move(p_right.m_dataHolder))
 {
+    p_right.m_data = nullptr;
+    p_right.m_length = 0;
 }
 
-
 ByteArray::ByteArray(std::uint8_t* p_array, std::size_t p_length, bool p_transferOnwership)
-    : m_data(p_array),
-      m_length(p_length)
+    : m_data(p_array)
+    , m_length(p_length)
 {
     if (p_transferOnwership)
     {
         m_dataHolder.reset(m_data, std::default_delete<std::uint8_t[]>());
     }
 }
 
-
-ByteArray::ByteArray(std::uint8_t* p_array, std::size_t p_length, std::shared_ptr<std::uint8_t> p_dataHolder)
-    : m_data(p_array),
-      m_length(p_length),
-      m_dataHolder(std::move(p_dataHolder))
+ByteArray::ByteArray(std::uint8_t* p_array, std::size_t p_length, std::shared_ptr<std::uint8_t> p_dataHolder) noexcept
+    : m_data(p_array)
+    , m_length(p_length)
+    , m_dataHolder(std::move(p_dataHolder))
 {
 }
 
-
-ByteArray::ByteArray(const ByteArray& p_right)
+ByteArray::ByteArray(const ByteArray& p_right) noexcept
     : m_data(p_right.m_data),
       m_length(p_right.m_length),
       m_dataHolder(p_right.m_dataHolder)
 {
 }
 
-
 ByteArray&
-ByteArray::operator= (const ByteArray& p_right)
+ByteArray::operator=(const ByteArray& p_right) noexcept
 {
-    m_data = p_right.m_data;
-    m_length = p_right.m_length;
-    m_dataHolder = p_right.m_dataHolder;
+    if (this != std::addressof(p_right))
```

Comment:
I think shared_ptr has already handled such case? Not sure if this is necessary.

Reply:
I haven't checked yet but it is just a recommended standard check in both copy constructor and move constructors

Reply:
I take some investigation and think this check is not necessary for std::shared_ptr and most other stl containers (it seems a popular implementation for shared_ptr assign function is to create a temp var and then swap it with `this`). So please remove this check.

Reply:
I have reverted back the checks